### PR TITLE
net-misc/youtube-viewer: Bump to 3.2.7 release

### PR DESCRIPTION
**Pull request by zpuskas**
Adjusted dependencies according to release, fixed RDEPEND of video players,
allowed ffmpeg with gnutls

Gentoo Bug: #600460

Package-Manager: Portage-2.3.6, Repoman-2.3.2

### CODE REVIEW

File: net-misc/youtube-viewer/youtube-viewer-3.2.7.ebuild [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+		virtual/freedesktop-icon-theme
+		x11-libs/gdk-pixbuf:2[X,jpeg]
+	)
+	|| ( media-video/ffmpeg[openssl] media-video/ffmpeg[gnutls] )
+	|| ( media-video/mpv media-video/mplayer media-video/vlc gtk? ( media-video/smplayer ) )"
+DEPEND="dev-perl/Module-Build"
+
+SRC_TEST="do"
+
+src_prepare() {
```
Comment (mgorny) [inline net-misc/youtube-viewer/youtube-viewer-3.2.7.ebuild]: Seems equivalent to no `src_prepare()`. Any reason to call it explicitly?

File: net-misc/youtube-viewer/youtube-viewer-3.2.7.ebuild [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+	perl-module_src_prepare
+}
+
+# build system installs files on "perl Build.PL" too
+# do all the work in src_install
```
Comment (mgorny) [inline net-misc/youtube-viewer/youtube-viewer-3.2.7.ebuild]: Have you considered submitting a patch upstream? ;-)

File: net-misc/youtube-viewer/youtube-viewer-3.2.7.ebuild [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+		dev-perl/File-ShareDir
+		>=dev-perl/Gtk2-1.244.0
+		virtual/freedesktop-icon-theme
+		x11-libs/gdk-pixbuf:2[X,jpeg]
+	)
+	|| ( media-video/ffmpeg[openssl] media-video/ffmpeg[gnutls] )
+	|| ( media-video/mpv media-video/mplayer media-video/vlc gtk? ( media-video/smplayer ) )"
+DEPEND="dev-perl/Module-Build"
+
+SRC_TEST="do"
```
Comment (kentfredric) [inline net-misc/youtube-viewer/youtube-viewer-3.2.7.ebuild]: SRC_TEST="do" is redundant for perl-module in EAPI6 ( literally not even used )

File: net-misc/youtube-viewer/youtube-viewer-3.2.7.ebuild [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+	perl-module_src_install
+
+	if use gtk ; then
+		domenu share/gtk-youtube-viewer.desktop
+		doicon share/icons/gtk-youtube-viewer.png
+	fi
+}
+
+pkg_preinst() {
+	perl_set_version
```
Comment (kentfredric) [inline net-misc/youtube-viewer/youtube-viewer-3.2.7.ebuild]: What is this doing ? `perl_set_version` sets a bunch of ENV vars, and that is all, but they don't appear to be used anywhere.

Also, setting those ENV vars during pre-inst seems weird. There's probably a reason, but it should be documented.

File: net-misc/youtube-viewer/youtube-viewer-3.2.7.ebuild [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+
+EAPI=6
+
+inherit eutils gnome2-utils perl-module
+
+DESCRIPTION="A command line utility for viewing youtube-videos in Mplayer"
+HOMEPAGE="https://trizenx.blogspot.com/2012/03/gtk-youtube-viewer.html"
+SRC_URI="https://github.com/trizen/youtube-viewer/archive/${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="|| ( Artistic GPL-1+ )"
```
Comment (kentfredric) [inline net-misc/youtube-viewer/youtube-viewer-3.2.7.ebuild]: I assume this implies "The same as perl itself".

Worth pointing out that perl-module.eclass provisions the "same as perl itself" value for LICENSE

File: net-misc/youtube-viewer/youtube-viewer-3.2.7.ebuild [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+# do all the work in src_install
+src_configure() { :; }
+src_compile() { :; }
+
+src_install() {
+	local myconf
+	if use gtk ; then
+		myconf="--gtk-youtube-viewer"
+	fi
+	perl-module_src_configure
```
Comment (kentfredric) [inline net-misc/youtube-viewer/youtube-viewer-3.2.7.ebuild]: Why are these configure steps performed in install instead of configure? Seems weird.

**Comment by mgorny** [comment]
Pingz.

**Comment by zpuskas** [comment]
Fixed all issues, but in the mean time there was a new version released. Abandoning this PR and creating a new one for the new version.

**Pull request closed without merging**